forc-plugins/forc-doc/src/render/item/context.rs

```diff
@@ -309,23 +346,212 @@ impl ItemContext {
+                    }
+                }
+            }
+        })
+    }
+}
+impl Renderable for TyImplTrait {
+    fn render(self, render_plan: RenderPlan) -> Result<Box<dyn RenderBox>> {
+        let TyImplTrait {
+            trait_name,
+            impl_type_parameters: _,
+            trait_type_arguments: _,
+            items,
+            trait_decl_ref: _,
+            implementing_for,
+            ..
+        } = self;
+
+        let mut rendered_items = Vec::new();
+        for item in items {
+            rendered_items.push(item.render(render_plan.clone())?)
+        }
+
+        let impl_for = box_html! {
+                div(id=format!("impl-{}", trait_name.suffix.as_str()), class="impl has-srclink") {
+                a(href=format!("{IDENTITY}impl-{}", trait_name.suffix.as_str()), class="anchor");
+                h3(class="code-header in-band") {
+                    : "impl ";
+                    : trait_name.suffix.as_str(); // TODO: add links
+                    : " for ";
+                    : implementing_for.span.as_str();
+                }
+            }
+        }
+        .into_string()?;
+
+        Ok(box_html! {
+            // check if the implementation has methods
+            @ if !rendered_items.is_empty() {
+                details(class="swaydoc-toggle implementors-toggle") {
+                    summary {
+                        : Raw(impl_for);
+                    }
+                    div(class="impl-items") {
+                        @ for item in rendered_items {
+                            : item;
+                        }
+                    }
+                }
+            } else {
+                : Raw(impl_for);
+            }
+        })
+    }
+}
+impl Renderable for TyTraitItem {
+    fn render(self, render_plan: RenderPlan) -> Result<Box<dyn RenderBox>> {
+        let item = match self {
+            TyTraitItem::Fn(item_fn) => item_fn,
+            TyTraitItem::Constant(_) => unimplemented!("Constant Trait items not yet implemented"),
+        };
+        let method = render_plan.decl_engine.get_function(item.id());
+        let attributes = method.attributes.to_html_string();
+
+        let mut fn_sig = format!("fn {}(", method.name.as_str());
+        for param in &method.parameters {
+            let mut param_str = String::new();
+            if param.is_reference {
+                write!(param_str, "ref ")?;
+            }
+            if param.is_mutable {
+                write!(param_str, "mut ")?;
+            }
+            if param.is_self() {
+                write!(param_str, "self,")?;
+            } else {
+                write!(
+                    fn_sig,
+                    "{} {},",
+                    param.name.as_str(),
+                    param.type_argument.span.as_str()
+                )?;
+            }
+        }
+        write!(fn_sig, ") -> {}", method.return_type.span.as_str())?;
+        let multiline = fn_sig.chars().count() >= 60;
```

Comment:
nit: move this to a helper function since we do this in two places

Reply:
Right, I would have if I could have but the types are actually different even though they look the same and their fields don't always match up so I couldn't use a generic implementation. If you know of a way to go about this I don't mind making a follow up.